components/board/genericlinux/gpio.go

```diff
@@ -375,6 +381,10 @@ func (pin *gpioPin) SetPWMFreq(ctx context.Context, freqHz uint, extra map[strin
 	pin.mu.Lock()
 	defer pin.mu.Unlock()
 
+	if freqHz < 1 {
+		return errors.New("must set PWM frequency to a positive value")
+	}
+
```

Comment:
> The rpi4 doesn't validate that the frequency is positive because apparently setting it to 0 already sets it to the default of 800 Hz

just to confirm, theres no equivalent behavior for the generic linux setup right? cuz generic

Reply:
Correct!